searcher/analyse/analyse.go

```diff
@@ -0,0 +1,61 @@
+package analyse
+
+import (
+	"log"
+	"strings"
+
+	"github.com/sleepypioneer/resource-searcher/searcher/resource"
+)
+
+var (
+	keywords    = []string{"python", "webservice", "server", "Django", "list"}
+	punctuation = []string{".", ",", "!", ":", ";", "'", "/", "-", "_", "*", "&", "{", "}", "[", "]"}
+)
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+// WordCount returns total number of words in the article
+func WordCount(content []string) int {
+	wordCount := 0
+	// for each paragraph in the article we count how many words it contains.
+	for _, p := range content {
+		// remove all punction in paragraph
+		for _, punc := range punctuation {
+			p = strings.ReplaceAll(p, punc, "")
+		}
+		// split paragraph into words list
+		words := strings.Split(p, " ")
+		wordCount += len(words)
+	}
+	// return total word count for the article
+	return wordCount
+}
+
+// FindTopics retuns a list of topics which the article includes
+func FindTopics(content []string) (topics []string) {
+	for _, p := range content {
+		for _, keyword := range keywords {
+			if strings.Contains(p, keyword) && !stringInSlice(keyword, topics) {
+				topics = append(topics, keyword)
+			}
+		}
+	}
+	return topics
+}
+
+// Analyse runs the various analysing logic on the give document
+func Analyse(doc *resource.Document) {
```

Comment:
As `Analyse` modifies the `doc`, I'd suggest to make a method out of it (and move it to `package resource`), to make it more obvious:

```go
package resource

// Analyse runs various analysing steps on the given document and updates it.
func (doc *Document) Analyse() error {
	doc.WordCount = analyse.WordCount(doc.Content)
	doc.Topics = analyse.FindTopics(doc.Content)
}